src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -465,7 +465,7 @@ public NetworkPeer(NetworkAddress peerAddress, Network network, NetworkPeerConne
             parameters = parameters ?? new NetworkPeerConnectionParameters();
             this.Inbound = false;
             this.Behaviors = new NetworkPeerBehaviorsCollection(this);
-            this.MyVersion = parameters.CreateVersion(peerAddress.Endpoint, network);
+            this.MyVersion = parameters.CreateVersion(peerAddress.Endpoint, network, this.dateTimeProvider.GetTimeOffset());
```

Comment:
you're passing GetTimeOffset but the `CreateVersion` method was actually using  `DateTimeOffset.UtcNow` before.
Is that intentional?

Reply:
Yes, that is another thing, I'd like to get rid completely of DateTimeOffset from FN, but that's different thing than introducing date time provider. So now the idea was to replace every `DateTimeOffset.UtcNow` with `this.dateTimeProvider.GetTimeOffset()` and every `DateTime.Now` or `DateTime.UtcNow` to be replaced with `this.dateTimeProvider.GetUtcTime()`.

So at this point I want to preserve the same types, just use the provider interface for delivering the actual value.